demo/scripts/common.sh

```diff
@@ -55,3 +55,18 @@ SCRIPTS_DIR="$(cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)"
 # The name of the kind image to build / run
 : ${KIND_IMAGE:="kindest/node:${KIND_K8S_TAG}"}
 
+# Container tool, e.g. docker/podman
+if [[ -z "${CONTAINER_TOOL}" ]]; then
+    if [[ -n "$(which docker)" ]]; then
+        echo "Docker found in PATH."
+        CONTAINER_TOOL=docker
+    elif [[ -n "$(which podman)" ]]; then
+        echo "Podman found in PATH."
+        CONTAINER_TOOL=podman
+    else
+        echo "No container tool detected. Please install Docker or Podman."
+        return 1
+    fi
+fi
+
+: ${KIND:="env KIND_EXPERIMENTAL_PROVIDER=${CONTAINER_TOOL} kind"}
```

Comment:
just for my own understanding -- why is the `env` included here?

Reply:
Right, without it substituting `${KIND}` makes the shell treat `KIND_EXPERIMENTAL_PROVIDER=podman` as a command, causing something like `.../demo/scripts/create-kind-cluster.sh: line 30: KIND_EXPERIMENTAL_PROVIDER=podman: command not found`.

If you can think of a better solution, I'll be happy to use it.

Reply:
I guess the alternative would be to make sure that `KIND_EXPERIMENTAL_PROVIDER` is exported in every script where `kind` is called (and then go back to not having a `KIND` variable at all).

Reply:
To be clear, I don't feel too strongly either way so long as it works (and will work in all supported environments).

Reply:
I personally see no harm in using `env` as its purpose is exactly running a command with passed env variables https://www.gnu.org/software/coreutils/manual/html_node/env-invocation.html. It's also widely supported https://en.wikipedia.org/wiki/Env. On the other hand, it allows us to set the `KIND` variable in a single place, and keep it alongside other Kind-related variables.